discretenet/problems/gisp/gisp.py

```diff
@@ -0,0 +1,216 @@
+        which_set="SET2",
+        graph_instance=None,
+        min_n=100,
+        max_n=100,
+        er_prob=0.1,
+        set_param=100.0,
+        alpha=0.75,
+    ):
+        """
+        Initialize the GISP generator instance following https://doi.org/10.1016/j.ejor.2016.11.050
+        :param random_seed: The random seed to use
+        :param path_prefix:  Path prefix to pass to instance ``save()`` methods
+            during batch generation. Must be set as a public instance attribute,
+            to be changed by the user after generator instantiation if desired.
+        :param which_set: can take values “SET1” or “SET2". These are two types
+            of models for assigning costs/revenues to edges/nodes in the generated
+            instances.
+        :param graph_instance: path to .clq file of DIMACS graph or do not provide
+            it and a random graph will be generated.
+        :param min_n/max_n: min/max number of nodes a randomly generated
+            graph can have, if not using a DIMACS graph
+        :param er_prob: the probability of an edge existing in the randomly
+            generated graph, if not using a DIMACS graph.
+        :param set_param: a value that affects the node revenues and/or edge costs
+        :param alpha: probability of an edge being in the set of edges which are
+            legal to remove (at a cost)
+        """
+
+        super().__init__(random_seed, path_prefix)
+        random.seed(random_seed)  # should be moved into set_seed()
+        self.problem = None
+        self.which_set = which_set
+        self.graph_instance = graph_instance
+        self.min_n = min_n
+        self.max_n = max_n
+        self.er_prob = er_prob
+        self.set_param = set_param
+        self.alpha = alpha
+        self.E2 = set()
+
+        if self.graph_instance is None:
+            # Generate random graph
+            num_nodes = random.randint(self.min_n, self.max_n)
+            self.graph = nx.erdos_renyi_graph(
+                n=num_nodes, p=self.er_prob, seed=self.random_seed
+            )
+            self.name = "er_n=%d_m=%d_p=%.2f_%s_setparam=%.2f_alpha=%.2f" % (
+                num_nodes,
+                nx.number_of_edges(self.graph),
+                self.er_prob,
+                self.which_set,
+                self.set_param,
+                self.alpha,
+            )
+        else:
+            self.dimacs_to_nx()
+            self.name = "%s_%s_%g_%g" % (
+                self.graph_instance,
+                self.which_set,
+                self.alpha,
+                self.set_param,
+            )
+
+    def dimacs_to_nx(self):
+        g = nx.Graph()
+        with open_text("discretenet.problems.gisp.graphs", self.graph_instance) as f:
+            for line in f:
+                arr = line.split()
+                if line[0] == "e":
+                    g.add_edge(int(arr[1]), int(arr[2]))
+        self.graph = g
+
+    def problem(self) -> Problem:
```

Comment:
Don't need this anymore, I replaced it with the `Generator[GISPProblem]`

Reply:
Updated